### PR TITLE
Resume background audio and voiceover after voice note or playing media

### PR DESCRIPTION
**Pull request by michaelkirk**
Based on #2977 so merge that first.

PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/ViewControllers/CallViewController.swift [hunk 1]
```diff
@@ -961,6 +962,29 @@ class CallViewController: OWSViewController, CallObserver, CallServiceObserver {
...
+        // Which sources are available depends on the state of your Session.
+        // When the audio session is not yet in PlayAndRecord none are available
+        // Then if we're in speakerphone, bluetooth isn't available.
+        // So we accrue all possible audio sources in a set, and that list lives as longs as the CallViewController
+        // The downside of this is that if you e.g. unpair your bluetooth mid call, it will still appear as an option
+        // until your next call.
+        // FIXME: There's got to be a better way, but this is where I landed after a bit of work, and seems to work
+        // pretty well in practice.
+        let availableInputs = callAudioService.availableInputs
+        self.allAudioSources.formUnion(availableInputs)
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/CallViewController.swift]: Seems like we'll need to test this on iOS 8, 9, 10, 11.  Please make sure to specify that in the test plan.

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 2]
```diff
@@ -3279,20 +3276,21 @@ - (void)endRecordingVoiceMemo
...
 {
     OWSAssertIsOnMainThread();
 
     DDLogDebug(@"cancelRecordingVoiceMemo");
 
-    [self resetRecordingVoiceMemo];
-}
-
-- (void)resetRecordingVoiceMemo
-{
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: Huh, isn't this `OWSAssertIsOnMainThread` twice in a row?

File: Signal/src/ViewControllers/CallViewController.swift (hunk 1 above)
Comment (michaelkirk) [reply Signal/src/ViewControllers/CallViewController.swift]: Sure thing.

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m (hunk 2 above)
Comment (michaelkirk) [reply Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: Removed the redundant assert - thanks!

**Comment by michaelkirk** [comment]
Squashed into c646f763

**Pull request closed without merging**